frontstage/controllers/notify_controller.py

```diff
@@ -41,26 +41,27 @@ def _send_message(self, email, template_id, personalisation=None, reference=None
             logger.info("Notification not sent. Notify is disabled.")
             return
 
-        try:
-            notification = {
-                "emailAddress": email,
-            }
-            if personalisation:
-                notification.update({"personalisation": personalisation})
-            if reference:
-                notification.update({"reference": reference})
+        notification = {
+            "emailAddress": email,
+        }
+        if personalisation:
+            notification.update({"personalisation": personalisation})
+        if reference:
+            notification.update({"reference": reference})
 
-            url = urlparse.urljoin(self.notify_url, str(template_id))
-            auth = app.config['SECURITY_USER_NAME'], app.config['SECURITY_USER_PASSWORD']
-            response = requests.post(url, json=notification, auth=auth,
-                                     timeout=int(app.config['REQUESTS_POST_TIMEOUT']))
+        url = urlparse.urljoin(self.notify_url, str(template_id))
+        auth = app.config['SECURITY_USER_NAME'], app.config['SECURITY_USER_PASSWORD']
+        response = requests.post(url, json=notification, auth=auth,
+                                 timeout=int(app.config['REQUESTS_POST_TIMEOUT']))
+        status_code = response.status_code
 
+        if 201 <= status_code <= 399:
             logger.info('Notification id sent via Notify-Gateway to GOV.UK Notify.', id=response.json()["id"])
-
-        except Exception as e:
+        else:
             ref = reference if reference else 'reference_unknown'
             raise exceptions.RasNotifyError("There was a problem sending a notification to Notify-Gateway "
```

Comment:
You don't need to pass these variables in as fstrings, `RasNotifyError` takes kwargs
```suggestion
            raise RasNotifyError("There was a problem sending a notification to Notify-Gateway.", url=url, status_code=status_code, message=response.text, reference=ref)
```